manager/src/main/java/com/redhat/service/bridge/manager/api/internal/ShardBridgesSyncAPI.java

```diff
@@ -86,9 +88,13 @@ public Response getProcessors() {
         String shardId = identityResolver.resolve(jwt);
         failIfNotAuthorized(shardId);
         LOGGER.info("Request from Shard for Processors to deploy or delete.");
-        List<Processor> processorToDeployOrDelete = processorService.getProcessorByStatusesAndShardIdWithReadyDependencies(SHARD_REQUESTED_STATUS, shardId);
-        LOGGER.info("Found {} processor(s) to deploy or delete", processorToDeployOrDelete.size());
-        return Response.ok(processorToDeployOrDelete
+        List<Processor> processorToDeploy = processorService.getProcessorsToDeployByShardIdWithReadyDependencies(shardId);
+        List<Processor> processorToDelete = processorService.getProcessorsToDeleteByShardIdWithDeletedDependencies(shardId);
```

Comment:
Why two different queries with following merge of results? Is it possible for this code to produce some race condition (the status of the DB changes between the two queries)?

Reply:
I lean towards a single query and method on the `ProcessorService` e.g `ProcessorService.getProcessorWorkForShard(String shardId)` which just returns a list of the work that the given Shard needs to do. I also think the query you're proposing `...and ((p.status='ACCEPTED' and p.dependencyStatus='READY' or (p.status='DEPROVISION' and p.dependencyStatus='DELETED'))` is pretty readable IMHO

That  encapsulates better within the `ProcessorService` the way we determine what a Shard needs to do and should we ever change that logic, `ShardBridgesSyncAPI` doesn't need to change.

Reply:
Re: @danielezonca comment on concurrent operations on a Resource, we have https://issues.redhat.com/projects/MGDOBR/issues/MGDOBR-398 as part of this Epic to work on. I suggest we get this PR merged to get us back to our current state and then we iterate again 👍

Reply:
Fine for me to proceed 👍 
I think we should revisit `ShardBridgesSyncAPI` to provide "the full" status of the shard to avoid misalignments (i.e. for some reason a resource is deleted on the Shard and we need to make sure this is recreated) but it is a different topic/task 👍

Reply:
Restoring to the single read.....

Reply:
New commit pushed restoring a single query.